src/vmm/src/vstate/vcpu/x86_64.rs

```diff
@@ -392,6 +461,34 @@ pub struct VcpuState {
     vcpu_events: kvm_vcpu_events,
     xcrs: kvm_xcrs,
     xsave: kvm_xsave,
+    #[version(start = 2, default_fn = "default_tsc_khz", ser_fn = "ser_tsc")]
+    tsc_khz: Option<u32>,
+}
+
+impl VcpuState {
+    fn default_tsc_khz(_: u16) -> Option<u32> {
+        warn!("CPU TSC freq not found in snapshot");
+        None
+    }
+
+    fn ser_tsc(&mut self, target_version: u16) -> VersionizeResult<()> {
+        if target_version < 2 {
+            // v0.24 and older versions do not support TSC scaling.
+            warn!(
+                "Saving to older snapshot version, TSC freq not included \
+                in snapshot."
+            );
+        } else if self.tsc_khz.is_none() {
+            // v0.25 and newer snapshots without TSC will only work on
+            // the same CPU model as the host on which they were taken.
+            warn!(
+                "TSC freq not available. Snapshot will only work on the \
```

Comment:
I would rephrase: `Snapshot cannot be loaded on a different CPU model.`

Reply:
Done.